src/custom/utils/calculatePriceImpact.test.ts

```diff
@@ -1,53 +1,67 @@
 import { ChainId, WETH } from '@uniswap/sdk'
-import { CurrencyAmount, Percent, Token } from '@uniswap/sdk-core'
-import BigNumber from 'bignumber.js'
+import { CurrencyAmount, Token, TradeType } from '@uniswap/sdk-core'
 import { parseUnits } from 'ethers/lib/utils'
-
-function _calculateAbaPriceImpact(initialValue: string, finalValue: string) {
-  const initialValueBn = new BigNumber(initialValue)
-  const finalValueBn = new BigNumber(finalValue)
-  // TODO: use correct formula
-  // ((IV - FV) / IV / 2) * 100
-  const [numerator, denominator] = initialValueBn.minus(finalValueBn).div(initialValueBn).div('2').toFraction()
-
-  return new Percent(numerator.toString(), denominator.toString())
-}
+import { calculateFallbackPriceImpact } from './price'
 
 const WETH_MAINNET = new Token(ChainId.MAINNET, WETH[1].address, 18)
 const DAI_MAINNET = new Token(ChainId.MAINNET, '0x6b175474e89094c44da98b954eedeac495271d0f', 18)
 
 describe('A > B > A Price Impact', () => {
-  const AB_IN = parseUnits('1', WETH_MAINNET.decimals).toString()
-  const AB_OUT = parseUnits('1000', DAI_MAINNET.decimals).toString()
-
-  const abIn = CurrencyAmount.fromRawAmount(WETH_MAINNET, AB_IN)
-  const abOut = CurrencyAmount.fromRawAmount(DAI_MAINNET, AB_OUT)
-
   describe('[SELL] WETH --> DAI', () => {
     it('A > B > A SELL return proper price impact', () => {
-      // GIVEN a 1 WETH >> 1000 DAI AB Trade
-      // GIVEN a 1000 DAI >> 0.5 WETH BA Trade
-      const BA_OUT = parseUnits('0.5', WETH_MAINNET.decimals).toString()
-      const baOut = CurrencyAmount.fromRawAmount(WETH_MAINNET, BA_OUT)
+      // GIVEN a 100 WETH >> 80 DAI AB Trade
+      const abInParsed = parseUnits('100', WETH_MAINNET.decimals).toString()
```

Comment:
All this test follow the same structure


probably nice to define all cases in a simpler way (a const with the minimal data:
```
const ABA_CASES = [
   {
       soldAmount: 100, 
       receivedAmount: 80,
       recoveredAmount: 56,
       result: '25'
   }, ...
]
```

This way, is easier to review, create new cases, and u can even create `it` cases with dynamic names, so you know exactly the case that succeeds and/or fails